### DISCUSSION TITLE
Feature Request: Save As Preset

### DISCUSSION THREAD

**Discussion by austince** [gTile/gTile#383] [Ideas]
Hi there, really love the extension. 

I'd love to see a feature that let you save a configuration from the top-icon grid chooser (which is awesome to use) as a preset. If I have time in the coming weeks I'll attempt a PR but thought I'd throw the idea out there before to see what the community thinks.

I envision it working like this: 
- User selects window
- User clicks top icon
- User either checks or leaves unchecked (default) an option to `Save As Preset` 
- If checked, a drop-down appears with the Preset slots, with indicators to show if they have already been customized / their current value
- After finishing, the window selection is persisted to the selected preset

Using the grid tool might also be a good way to create Presets in the first place, and in that case, perhaps the `Save As` feature would not be required.

**Comment by scherepanov** [comment]
Nice idea, and definitely can be implemented as described.
It will work on current grid size only (out of predefined grid sizes). You can cycle through grid sizes with Space, when gTile is activated.
In fact, all pre-installed presets are on grid sizes that is not in pre-installed grid sizes.
In current gTile, if you want to add new grid size in configuration, you need to do it in config dialog (may be even need to logout/login after change).
You need to look more on logic how your new functionality will work with different grid size that is not in list.
BTW you can activate gTile not only by clicking on icon - you can also use keyboard shortcut  "Super-KP Enter"

**Comment by YoungElPaso** [comment]
Another option would be to show the 'last used' settings.  Maybe have a dialogue that shows the last 3 layouts selected via the grid? I'm not sure how that would work with multiple windows tho.
